File: easy/chevaux_de_course.rb
p = []
n = gets.to_i
n.times do
  pi = gets.to_i
  p.push(pi)
end
p = p.sort
ds = []
p.each_with_index do |pi, index|
    if index < p.length - 1
        d = p[index + 1] - pi
        ds.push(d)
    end
end
puts ds.min

File: easy/Chevaux_de_course.py
import sys
import math

p = []
n = int(input())
for i in range(n):
    pi = int(input())
    p.append(pi)

ds = []
p.sort()

for i in range(0, len(p) - 1, 1):
    d = p[i + 1] - p[i]
    ds.append(d)

print(min(ds))
